src/styles/calendar.scss

```diff
@@ -0,0 +1,78 @@
+.calendar {
+  margin: auto;
+  width: $calendar-width;
+  padding: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  transform: translateY(10%);
+
+  &--start-day {
+
+    $days:
+      "mon" 0,
+      "tue" 1,
+      "wed" 2,
+      "thu" 3,
+      "fri" 4,
+      "sat" 5,
+      "sun" 6;
+  
+    @each $name, $count in $days {
+      &--#{$name} &__day:first-child {
+        margin-left: ($cell-size * $count) + ($days-gap * $count);
+      }
+    }
+  }
+  
+  $days:
+    "mon" 0,
+    "tue" 1,
+    "wed" 2,
+    "thu" 3,
+    "fri" 4,
+    "sat" 5,
+    "sun" 6;
+
+  @each $name, $count in $days {
+    &--start-day--#{$name} &__day:first-child {
+      margin-left: ($cell-size * $count) + ($days-gap * $count);
+    }
+  }
+  
+  @for $date from 28 through 31 {
+    &--month-length--#{$date} :nth-child(n + #{$date + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    min-height: $cell-size;
+    min-width: $cell-size;
+    font-family: "Arial", sans-serif;
+    font-size: 30px;
+    justify-content: center;
+    align-items: center;

```

Comment:
Please pay attention to your CSS properties order. Check yourself everywhere
https://9elements.com/css-rule-order/